### PR TITLE
More Combat

### PR DESCRIPTION
**Pull request by danielwiktorczyk**
Combat Rounds, MulitTargeting, and ThunderCloud Tweaks

### CODE REVIEW

File: Assets/Scripts/Combat/Attack.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (eli-lam) [inline Assets/Scripts/Combat/Attack.cs:5]: What are these imports used for here?

**Review by eli-lam — APPROVED** [review]
Looks good to me, just one small comment that can be addressed later

File: Assets/Scripts/Combat/Attack.cs (hunk 1 above)
Comment (armandorusso) [reply Assets/Scripts/Combat/Attack.cs:5]: Generic Library: We used it for making the list of enemies, allies, and sorting who goes first. 
Linq Library: Used for lambda functions and sorting (Link: https://unity3d.college/2017/07/01/linq-unity-developers/). We used this for turn priority and targeting mostly.
Threading: We used it to just sleep the program when selecting an ability and when the enemies attack. This is temporary to just simulate combat (and not make each move instant).

File: Assets/Scripts/Combat/CombatSystem.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                return false;
+
+        Debug.Log("!!!!Allies Win!!!!");
+        return true;
+    }
+
+    private bool EnemiesWon()
+    {
+        foreach (GameObject ally in AllyCombatants)
+            if (ally.GetComponent<EnemyCombatant>().isAlive)
```
Comment (armandorusso) [inline Assets/Scripts/Combat/CombatSystem.cs:87]: Should be ally.GetComponent<AllyCombatant>().isAlive
What you wrote there is checking for an EnemyCombatant component, which doesn't exist on an ally

**Review by armandorusso — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Combat/Combantants/EnemyCombatant.cs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+        StartCoroutine(SimpleEnemyAttack()); // 
+        //Thread.Sleep(2000);
+    }
+
+    IEnumerator SimpleEnemyAttack()
+    {
+        yield return new WaitForSeconds(2);
+
+        Attack attack = new Attack(100);
```
Comment (armandorusso) [inline Assets/Scripts/Combat/Combantants/EnemyCombatant.cs:74]: The enemy will still attack, even if it's dead. Add a check to see if it's alive and if it is, attack.

**Review by armandorusso — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Combat/Attack.cs (hunk 1 above)
Comment (danielwiktorczyk) [reply Assets/Scripts/Combat/Attack.cs:5]: We are using these, but not in the file. Removing now!

File: Assets/Scripts/Combat/CombatSystem.cs (hunk 2 above)
Comment (danielwiktorczyk) [reply Assets/Scripts/Combat/CombatSystem.cs:87]: Fixed! Good catch

File: Assets/Scripts/Combat/Combantants/EnemyCombatant.cs (hunk 3 above)
Comment (danielwiktorczyk) [reply Assets/Scripts/Combat/Combantants/EnemyCombatant.cs:74]: Added in the Base class!

**Review by armandorusso — APPROVED** [review]
Saw fixes on call with screen share. Good job!

**Comment by danielwiktorczyk** [comment]
> Saw fixes on call with screen share. Good job!

No you, we peer programmed haha

**Pull request merged**